C++ StateFeedbackGainInterface

wxmerkt opened the pull request:
Work in progress branch for integration

cmastalli commented on include/whole_body_mpc_msgs/conversions.h:
```diff
@@ -0,0 +1,72 @@
+//
+// Copyright (c) 2021, University of Oxford
```
Can we use a short format instead of the whole message?
This is the example
```c++
///////////////////////////////////////////////////////////////////////////////
// BSD 3-Clause License
//
// Copyright (C) 2021, University of Oxford
// Copyright note valid unless otherwise stated in individual files.
// All rights reserved.
///////////////////////////////////////////////////////////////////////////////
```

We should also add the include the LICENSE file.

cmastalli commented on include/whole_body_mpc_msgs/conversions.h:
```diff
@@ -0,0 +1,72 @@
+/**
+ * @brief Conversion of Eigen::MatrixXd to message for a given whole_body_mpc_msgs::StateFeedbackGain message reference
+ *
+ * @param msg whole_body_mpc_msgs::StateFeedbackGain (reference, will be modified)
```
This way of documenting is strange to me, what I often do is:
```c++
* @param[out] msg    Message of the state feedback gain to be modified
```

So, the second argument, after msg, is the item description (not the type of argument)

cmastalli reviewed (commented):
I just provide a few comments.

Ping me when the PR is finished.

wxmerkt commented:
Thank you for looking over the PR - I have addressed the comments. This code is now used in the interface as in the other repository. I am not planning further modifications here right now (it's minimal and working) and will remove the "WIP".

cmastalli commented on include/whole_body_mpc_msgs/conversions.h:
```diff
@@ -0,0 +1,57 @@
+/**
+ * @brief Conversion of Eigen::MatrixXd to message for a given whole_body_mpc_msgs::StateFeedbackGain message reference
+ *
+ * @param[out] msg whole_body_mpc_msgs::StateFeedbackGain (reference, will be modified)
```
I think you missed this line

cmastalli commented on include/whole_body_mpc_msgs/conversions.h:
```diff
@@ -0,0 +1,57 @@
+ * @brief Conversion of a state feedback gain from a whole_body_mpc_msgs::StateFeedbackGain message to an
+ * Eigen::MatrixXd
+ *
+ * @param[in] msg whole_body_mpc_msgs::StateFeedbackGain
```
and this one

cmastalli commented on include/whole_body_mpc_msgs/state_feedback_gain_interface.h:
```diff
@@ -0,0 +1,66 @@
+   * @brief Conversion of Eigen::MatrixXd to message, returns a whole_body_mpc_msgs::StateFeedbackGain message
+   *
+   * @param[in] K state feedback gain (size nu * nx)
+   * @return whole_body_mpc_msgs::StateFeedbackGain
```
same here.
Instead, I will document it as
```cpp
* @return The state feedback gain
```

cmastalli commented on include/whole_body_mpc_msgs/state_feedback_gain_interface.h:
```diff
@@ -0,0 +1,66 @@
+   * @brief Conversion of whole_body_mpc_msgs::StateFeedbackGain message to Eigen::MatrixXd to message, returns a
+   * Eigen::MatrixXd
+   *
+   * @param[in] msg whole_body_mpc_msgs::StateFeedbackGain
```
also here

cmastalli commented on CMakeLists.txt:
```diff
@@ -7,45 +7,41 @@ SET(PROJECT_DESCRIPTION "Whole-body MPC messages")
-  SET_PROPERTY(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
-ENDIF()
+MESSAGE(STATUS "Copyright (C) 2020 University of Edinburgh")
+MESSAGE(STATUS "All rights reserved.")
```
Did you miss this? or do you prefer to not include Oxford?

cmastalli reviewed (commented):
I included other comments. Sorry for not being specific enough.
We should also push the LICENSE file, we could copy the same one used in Crocoddyl.

cmastalli commented:
@wxmerkt  -- I will handle now these tiny changes.

cmastalli commented:
Thanks @wxmerkt :)

wxmerkt commented:
Thank you - I was planning to pick it up this afternoon but happy to see it's handled :-)
